sandbox/grommet-app/src/pages/layouts/kinds/OpsRampDetail/OpsRampDetailTable.tsx

```diff
@@ -0,0 +1,207 @@
+    }
+    if (view && view.properties && typeof view.properties === 'object') {
+      Object.keys(view.properties).forEach(property => {
+        if (view.properties && Array.isArray(view.properties[property])) {
+          filteredData = filteredData.filter(
+            nodes =>
+              view.properties &&
+              view.properties[property].includes(nodes[property]),
+          );
+        }
+      });
+    }
+
+    setResult(filteredData);
+    setTotal(filteredData.length);
+  }, [view, quickFilter]);
+
+  const columns: ColumnConfig<Node>[] = [
+    {
+      property: 'name',
+      pin: true,
+      primary: true,
+      header: 'Name',
+      render: datum => (
+        <Box align="center" direction="row" gap="xsmall">
+          {datum.state === 'up' ? (
+            <StatusGoodSmall color="status-ok" />
+          ) : datum.state === 'down' ? (
+            <StatusCriticalSmall color="status-critical" />
+          ) : datum.state === 'unknown' ? (
+            <StatusUnknownSmall color="status-unknown" />
+          ) : datum.state === 'undefined' ? (
+            <StatusPlaceholderSmall />
+          ) : null}
+          <Anchor
+            onClick={() => {
+              setShowResultDetails(true);
+            }}
+          >
+            {datum.name}
+          </Anchor>
+        </Box>
+      ),
+    },
+    {
+      property: 'ip address',
+      header: 'IP Address',
+      sortable: false,
+    },
+    {
+      property: 'make',
+      header: 'Make',
+      sortable: false,
+    },
+    {
+      property: 'model',
+      header: 'Model',
+      sortable: false,
+    },
+  ];
+
+  return (
+    <Data
+      data={result}
+      total={total}
+      filteredTotal={result.length}
+      defaultView={defaultView}
+      gap="medium"
+      properties={{
+        model: {
+          label: 'Model',
+          options: ['VirtualMachine', 'VMware Virtual Platform'],
+        },
+        state: {
+          label: 'State',
+          options: ['up', 'down', 'unknown', 'undefined'],
+        },
+      }}
+      view={view}
+      onView={setView}
+    >
+      <QuickFilters
+        value={quickFilter}
+        setValue={setQuickFilter}
+        counts={{ up, down, unknown, undefinedState }}
+      />
+      <Box>
+        <Toolbar>
+          <DataSearch />
+          <DataFilters layer />
+          <DataSort drop />
+          <Button kind="toolbar" icon={<Share />} />
+          <Button kind="toolbar" icon={<SettingsOption />} />
+        </Toolbar>
+        <DataSummary />
+        <Box overflow="auto">
+          <DataTable
+            aria-describedby="node-table"
+            onSelect={() => {}}
+            columns={columns}
+            sortable
+            data={result.slice(
+              ((view.page ?? 1) - 1) * (view.step ?? 10),
+              (view.page ?? 1) * (view.step ?? 10),
+            )}
+          />
+        </Box>
+        <Pagination
```

Comment:
I'd expect `step` and `numItems` coming from view and current view's filtered items length.

Reply:
working on this now

Reply:
Im not sure Im following what you mean on this comment.

Reply:
Possibly related to: https://github.com/grommet/hpe-design-system/pull/4796/files#r1992433860 and removing `<DataTable data ...`.

Can leave as is and see if we are getting the correct behavior after these changes.